server/sonar-server/src/main/java/org/sonar/server/permission/PermissionUpdater.java

```diff
@@ -177,18 +182,18 @@ private Long getTargetedGroup(DbSession session, String group) {
     }
   }
 
-  private static boolean shouldSkipPermissionChange(Operation operation, List<String> existingPermissions, PermissionChange permissionChange) {
+  private static boolean shouldSkipPermissionChange(Operation operation, Collection<String> existingPermissions, PermissionChange permissionChange) {
     return (Operation.ADD == operation && existingPermissions.contains(permissionChange.permission())) ||
       (Operation.REMOVE == operation && !existingPermissions.contains(permissionChange.permission()));
   }
 
   @CheckForNull
-  private Long getComponentId(DbSession session, @Nullable String componentKey) {
+  private ComponentDto getComponent(DbSession session, @Nullable String componentKey) {
     if (componentKey == null) {
       return null;
     } else {
       ComponentDto component = componentFinder.getByKey(session, componentKey);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "component". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1488)